doc/api/async_hooks.md

```diff
@@ -973,6 +973,21 @@ chains. It allows storing data throughout the lifetime of a web request
 or any other asynchronous duration. It is similar to thread-local storage
 in other languages.
 
+## Difference with raw Async Hooks
+
+`AsyncLocalStorage` doesn't provide any new functionality that raw Async Hooks
+doesn't already provide: anything implemented with `AsyncLocalStorage` can be
+implemented with Async Hooks instead.
```

Comment:
I would probably clarify that the context object management aspect is not a feature of Async Hooks but rather that it could be implemented on top of it. That makes it more clear that `AsyncLocalStorage` is providing some actual _functionality_ that does not otherwise exist without building it manually.